Snail/Observable.swift

```diff
@@ -155,7 +155,7 @@ public class Observable<T>: ObservableType {
 
         let semaphore = DispatchSemaphore(value: 0)
 
-        subscribe(onNext: { value in
+        _ = subscribe(onNext: { value in
```

Comment:
I think we should add this to the subscribers array right? This way we remove this subscription when `deinit`